jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -24,9 +24,58 @@
 from jdaviz.core.tools import ICON_DIR
 from jdaviz.core.custom_traitlets import IntHandleEmpty
 
+from scipy.interpolate import make_interp_spline
+
 __all__ = ['PlotOptions']
 
 
+class SplineStretch:
+    """
+    A class to represent spline stretches.
+
+    Attributes:
+        k (int): Degree of the smoothing spline.
+        bc_type (str or None): Boundary condition type.
+        t (array-like or None): Array of knot positions.
+        x (array-like): The x-coordinates of the data points.
+        y (array-like): The y-coordinates of the data points.
+        spline (object): Interpolating spline.
+    """
+
+    def __init__(self):
+        """
+        Initializes the SplineStretch class with default knot values.
+        """
+        self.k = 3
+        self.bc_type = None
+        self.t = None
+        self.update_knots(
+            x=np.array([0, 0.1, 0.2, 0.7, 1]),
+            y=np.array([0, 0.05, 0.3, 0.9, 1])
+        )
+
+    def __call__(self, values, out=None, clip=False):
+        # TODO: Review the 'out' and 'clip' parameters once glue upstream is updated.

```

Comment:
Please cross-link the "glue upstream" issue URL here.

Reply:
Hmm then what do you mean by "glue upstream" here?

Reply:
Maybe the comment can be replaced with something along the lines of "for our uses, we can ignore `out` and `clip`, but those would need to be implemented before contributing this class upstream"?